From f06ce63ccd5255d00a5a28d0ba4e4d1f7b17ebe1 Mon Sep 17 00:00:00 2001
From: sliow <shi.jie.donavan.liow@intel.com>
Date: Mon, 24 Feb 2020 15:39:05 +0800
Subject: [PATCH 1/2] Revert "HACK: scsi: ufs: Add module parameters max_gear,
 dflt_hs_rate and dflt_hs_mode"

This reverts commit 75d616a930481c43ad9c6326ea76710328d3f8e5.
---
 drivers/scsi/ufs/ufshcd.c | 30 +++---------------------------
 1 file changed, 3 insertions(+), 27 deletions(-)

diff --git a/drivers/scsi/ufs/ufshcd.c b/drivers/scsi/ufs/ufshcd.c
index d3a2421..6a52d97 100644
--- a/drivers/scsi/ufs/ufshcd.c
+++ b/drivers/scsi/ufs/ufshcd.c
@@ -241,10 +241,6 @@ static struct ufs_dev_fix ufs_fixups[] = {
 	END_FIX
 };
 
-static int max_gear;
-static int dflt_hs_rate;
-static int dflt_hs_mode;
-
 static void ufshcd_tmc_handler(struct ufs_hba *hba);
 static void ufshcd_async_scan(void *data, async_cookie_t cookie);
 static int ufshcd_reset_and_restore(struct ufs_hba *hba);
@@ -4071,15 +4067,9 @@ static int ufshcd_get_max_pwr_mode(struct ufs_hba *hba)
 	if (hba->max_pwr_info.is_valid)
 		return 0;
 
-	if (dflt_hs_mode != FAST_MODE && dflt_hs_mode != FASTAUTO_MODE)
-		dflt_hs_mode = FAST_MODE;
-
-	if (dflt_hs_rate != PA_HS_MODE_A && dflt_hs_rate != PA_HS_MODE_B)
-		dflt_hs_rate = PA_HS_MODE_B;
-
-	pwr_info->pwr_tx = dflt_hs_mode;
-	pwr_info->pwr_rx = dflt_hs_mode;
-	pwr_info->hs_rate = dflt_hs_rate;
+	pwr_info->pwr_tx = FAST_MODE;
+	pwr_info->pwr_rx = FAST_MODE;
+	pwr_info->hs_rate = PA_HS_MODE_B;
 
 	/* Get the connected lane count */
 	ufshcd_dme_get(hba, UIC_ARG_MIB(PA_CONNECTEDRXDATALANES),
@@ -4125,12 +4115,6 @@ static int ufshcd_get_max_pwr_mode(struct ufs_hba *hba)
 		pwr_info->pwr_tx = SLOW_MODE;
 	}
 
-	if (max_gear > 0 &&
-	    (pwr_info->gear_rx > max_gear || pwr_info->gear_tx > max_gear)) {
-		pwr_info->gear_rx = max_gear;
-		pwr_info->gear_tx = max_gear;
-	}
-
 	hba->max_pwr_info.is_valid = true;
 	return 0;
 }
@@ -8932,14 +8916,6 @@ int ufshcd_init(struct ufs_hba *hba, void __iomem *mmio_base, unsigned int irq)
 }
 EXPORT_SYMBOL_GPL(ufshcd_init);
 
-module_param(max_gear, int, 0444);
-module_param(dflt_hs_rate, int, 0444);
-module_param(dflt_hs_mode, int, 0444);
-
-MODULE_PARM_DESC(, "Maximum gear: 1, 2 , 3 ...");
-MODULE_PARM_DESC(, "Default high speed rate series : 1 (= rate A), 2 (= rate B)");
-MODULE_PARM_DESC(, "Default high speed power mode: 1 (= FAST), 4 (= FASTAUTO)");
-
 MODULE_AUTHOR("Santosh Yaragnavi <santosh.sy@samsung.com>");
 MODULE_AUTHOR("Vinayak Holikatti <h.vinayak@samsung.com>");
 MODULE_DESCRIPTION("Generic UFS host controller driver Core");
-- 
2.7.4

